### ISSUE TITLE
x:Bind to static property with one or two way mode is not supported

### ISSUE THREAD

**Issue by lextm** [unoplatform/uno#3261]
## Current behavior

UWP project can be built, but macOS project cannot when using a specific style of binding,
``` xaml
<Button Visibility="{x:Bind local:App.InteractionViewModel.LeftMarginLoaded, Mode=OneWay}">Test</Button>
```
The errors are,
```
Error
CS0826
No best type found for implicitly-typed array
xamlgeneratorbug.macOS
C:\Users\lextm\source\repos\xamlgeneratorbug\xamlgeneratorbug.macOS\obj\iPhoneSimulator\Debug\g\XamlCodeGenerator\MainPage_a1587810f827c1ff169df7f2da9b3945.g.cs
65
```
and
```
Error
CS1503
Argument 5: cannot convert from '?[]' to 'string[]'
xamlgeneratorbug.macOS
C:\Users\lextm\source\repos\xamlgeneratorbug\xamlgeneratorbug.macOS\obj\iPhoneSimulator\Debug\g\XamlCodeGenerator\MainPage_a1587810f827c1ff169df7f2da9b3945.g.cs
65
```
## Expected behavior

Both projects should compile without a problem.

## How to reproduce it (as minimally and precisely as possible)

1. Clone https://github.com/lextm/xamlgeneratorbug
1. Compile and run its UWP project to confirm it's working.
1. Compile its macOS project and observe the compilation errors.

## Environment

Nuget Package: Uno.UI
Package Version(s): 2.4.0

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [x] 2019 (version: )
- [ ] for Mac (version: )

## Anything else we need to know?
This bug seems to affect other platforms.

**Comment by jeromelaban** [comment]
Thanks for the report! 

Currently, one-way x:Bind to an instance property through a static property is not supported. You can workaround this by Adding an instance property on your page that points to the `App.InteractionViewModel`. It's not ideal, but this will allow to achieve the same result.

**Comment by lextm** [comment]
@jeromelaban The workaround of wrapping such static items (properties or methods) does work as expected.

**Comment by MartinZikmund** [comment]
This is now fixed

**Comment by wokhan** [comment]
Hi! Not sure it is? 🤔😁 
I had to use a local "proxy" property again to get this to work, if not I get a weird generated code, something along this:
```
__ctxt.global::Path.To.My.Variable
```
which obviously doesn't work.
The fun part is that I got this at the beginning of Uno for a UWP project, and I had to fix it the same way so I feel like I'm traveling in time 😅

I'm using the latest Uno version (4.6.0-dev.532).

Thanks

**Comment by jeromelaban** [comment]
@wokhan could you provide a more concrete sample for this particular behavior you're seeing? It may be something that broke recently.

**Comment by wokhan** [comment]
Sure! I just realised that while the title of this issue matches my problem, the description does not (and I'm using WinUI if that matters).
I should probably open a new one (and will if you tell me). 

In the meantime:

I've this propery in App.xaml.cs (which allows me to have an already "properly typed" App instance available instead of casting App.Current - an instance of the "Application" standard class - everytime in XAML):
`public static App Instance => (App)Current;`

My app class hosts a "CurrentProject" property, not directly relevant here.

In my XAML I reference it everywhere like this:
`{x:Bind qnlzr:App.Instance.CurrentProject.WhatEver}`

With the qnlzr namespace being defined as follows:
`xmlns:qnlzr="using:QuAnalyzer" `

When compiling on a Uno head, I get that for **TwoWay** bindings:
` System.InvalidOperationException: Unable to find member [global::QuAnalyzer] on type [QuAnalyzer.UI.Pages.Compare]`
(it then fails to generated anything)

While for **OneWay** bindings, I get an incorrect alias (CS7000) message, due to the following being generated:
`___ctx is global::QuAnalyzer.UI.Pages.Compare ___tctx ? (object)(___tctx.global::QuAnalyzer.App.Instance.CurrentProject.UseSingleMapping)........`

As you can see the alias is in the middle, as if it was a property of the current context (i.e. the current page).

Looks like the issue is the same for both, but since I cannot find the generated files anywhere (still in-memory I guess) for TwoWay bindings, I cannot confirm.

Of course the workaround of having a local getter (in the current page code) targeting App.Instance works, but I still think something is broken somehow :)

Thanks!

**Comment by jeromelaban** [comment]
Thanks for the update. I'm still not able to reproduce with the steps you provided. Can you create a sample application with what you described and attached here? It's likely that there some important detail missing here.

**Comment by wokhan** [comment]
Thank you for giving it a try, I know you've already a lot to manage and I really appreciate!
Here is a project which fails building on my setup (latest VS preview, .NET 6.0.9, I kept the default libs versions imported by the Uno project template, but on my original app I updated all those).

Thank you again.
 
[BindingIssueRepro.zip](https://github.com/unoplatform/uno/files/9759216/BindingIssueRepro.zip)
